soh/soh/Enhancements/randomizer/3drando/item_list.cpp

```diff
@@ -177,7 +177,7 @@ void ItemTable_Init() {                              // RandomizerGet
     //Generic Items
     itemTable[RECOVERY_HEART]                    = Item(RG_RECOVERY_HEART,                    Text{"Recovery Heart",                  "Coeur de Vie",                     "Corazón"},                                 ITEMTYPE_ITEM, GI_HEART,             false, &noVariable,     RECOVERY_HEART);
     itemTable[GREEN_RUPEE]                       = Item(RG_GREEN_RUPEE,                       Text{"Green Rupee",                     "Rubis Vert",                       "Rupia verde"},                             ITEMTYPE_ITEM, GI_RUPEE_GREEN,       false, &noVariable,     GREEN_RUPEE);
-    itemTable[GREG_RUPEE]                        = Item(RG_GREG_RUPEE,                        Text{"Greg Rupee",                      "Rubis Greg",                       "Rupia Greg"},                              ITEMTYPE_ITEM, GI_RUPEE_GREEN,       false, &noVariable,     GREG_RUPEE);
+    itemTable[GREG_RUPEE]                        = Item(RG_GREG_RUPEE,                        Text{"Greg Rupee",                      "Rubis Greg",                       "Rupia Greg"},                              ITEMTYPE_ITEM, GI_RUPEE_GREEN,       true,  &Greg,           GREG_RUPEE);
```

Comment:
Since Greg is marked as an advancement item, this means that Greg will prevent "barren/foolish" hints if Greg is not in logic and is in a foolish area.

Short term solution could be to add a check for Greg and GregInLogic into the `IsMajorItem()` in `3drando/item.hpp`